packages/frontend/core/src/modules/editor-settting/impls/global-state.ts

```diff
@@ -0,0 +1,35 @@
+import type { GlobalState } from '@toeverything/infra';
+import { Service } from '@toeverything/infra';
+import { map, type Observable } from 'rxjs';
+
+import type { EditorSettingProvider } from '../provider/editor-setting-provider';
+
+const storageKey = 'editor-setting';
+
+/**
+ * just for testing, vary poor performance
+ */
+export class GlobalStateEditorSettingProvider
+  extends Service
+  implements EditorSettingProvider
+{
+  constructor(public readonly globalState: GlobalState) {
+    super();
+  }
+  set(key: string, value: string): void {
```

Comment:
it seems there is no way to remove a k/v, though it might not be an issue for now

Reply:
Yes, there is no need to delete the settings yet, we can easily add it in the future